day_7/10_speckled_frogs.js

```diff
@@ -0,0 +1,23 @@
+class NurseryRhyme {
+  constructor(protagonist, singular, number)
+  {
+    this.protagonist = protagonist
+    this.singular = singular
+    this.number = number
+  }
+tellRhyme() {
+  for (var i = this.number; i >= 1; i--)
+  if (i >=3) {
+    console.log([i] + " " + this.protagonist + " sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there were " + (i - 1) + " " + this.protagonist);
+    } else if (i == 2) {
+      console.log([i] + " " + this.protagonist + " sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there was " + (i - 1) + " " +  this.singular);
+    } else {
+      console.log([i] + " " + this.singular + " sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there were " + "no more " + this.protagonist + "!");
+    }
+  }
+}
+
+var speckledFrog = new NurseryRhyme("speckled frogs", "speckled frog", 10)
```

Comment:
Great work on arriving at a solution to this problem! 

Interesting choice to use a class to set this up - you'll learn more about this in Mod 1, but classes can sometimes be a little heavy-handed for this particular use case. One thing I want you to review is how you can use **parameters** with a **function** to achieve the same thing you did with this class example! For context, functions are more lightweight than classes and in fact, classes are [actually functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) under the hood. 

This could be a great opportunity to refactor this into a function that uses parameters instead of setting up an entire class.